### PR TITLE
Update authentication-azure-ad.md

### PR DESCRIPTION
**Pull request by mshafiqmk**
The documentation for data builder API with azure ad authentication was incomplete and misleading. I wasted a lot of time trying to make it work, until I found out that I needed to include the X-MS-API-ROLE request header. This crucial detail was added in this PR, but it should have been there from the start.

### CODE REVIEW

**Review by Aniruddh25 — APPROVED** [review]
Thank you for your contributions! We really appreciate your time and effort to improve our documentation, and apologize for the inconvenience caused by incomplete docs.

File: data-api-builder/authentication-azure-ad.md [hunk 1]
```diff
@@ -254,3 +254,6 @@ The following steps configure [delegated permissions](/azure/active-directory/de
       9. Scope: `api://<APP_ID>/Endpoint.Access` (don't forget this or authentication fails)
 3. Select Get New Access Token, and sign in with your Azure AD tenant credentials.
    1. It's expected that you get a Consent screen. This is normal and you must agree to authenticate successfully.
+
+## X-MS-API-ROLE request header
+The API requires two headers for authentication: a bearer token and X-MS-API-ROLE. The value of X-MS-API-ROLE must match the role specified in the token. For example, if the token has the role **Sample.Role**, then X-MS-API-ROLE should also be **Sample.Role**.
```
Comment (seantleonard) [inline data-api-builder/authentication-azure-ad.md:259]:
```suggestion
Data API builder requires the header `X-MS-API-ROLE` to authorize requests using custom roles. The value of `X-MS-API-ROLE` must match a role specified in the token. For example, if the token has the role **Sample.Role**, then X-MS-API-ROLE should also be **Sample.Role**. For more information, see the [Authorization](./authorization.md#user-roles) article.
```

**Review by seantleonard — COMMENTED** [review]
Thank you for your contributions! 

I suggested additional changes to link to our authorization documentation detailing these requirements. Also, while authentication is accomplished using the token in the bearer token header, authorization requires the extra `x-ms-api-role` header.

**Comment by seesharprun** [comment]
Thanks for your contribution, @mshafiqmk 

The repo has changed significantly since this pull request, so I pulled your changes into a new pull request, so you care appropriately credited with your contribution. Thank you again for your patience and effort!

**Pull request closed without merging**